models/tiegcm/model_mod.f90

```diff
@@ -769,1225 +509,345 @@ end subroutine get_state_meta_data
-integer ::             &
-     &  secstart(3,mxseries),&! secondary history model start time(s)
-     &  secstop(3,mxseries), &! secondary history model stop time(s)
-     &  sechist(3,mxseries), &! secondary history disk write frequency
-     &  secsave(3,mxseries), &! secondary history file save frequency
-     &  mxhist_sech,         &! max number of histories per secondary file
-     &  sech_nbyte            ! 4 or 8: write real or double values to secondary file
-!
-! Namelist for read:
-namelist/tgcm_input/                                        &
-     &  label,tempdir,magvol,amievol,date,calday,step,dispose,    &
-     &  source,source_start,output,start,stop,hist,save,          &
-     &  secout,secstart,secstop,sechist,secsave,secflds,          &
-     &  potential_model,eddy_dif,dynamo,tide,tide2,tide3m3,       &
-     &  f107,f107a,power,ctpoten,bximf,byimf,bzimf,swvel,swden,al,&
-     &  kp,colfac,tideann,aurora,gpi_ncfile,gswm_mi_di_ncfile,    &
-     &  gswm_mi_sdi_ncfile,gswm_nm_di_ncfile,gswm_nm_sdi_ncfile,  &
-     &  mxhist_prim,mxhist_sech,msreten,ntask_lat,ntask_lon,      &
-     &  start_day,start_year,calendar_advance,see_ncfile,         &
-     &  ctpoten_time,power_time,bximf_time,byimf_time,bzimf_time, &
-     &  kp_time,al_time,swden_time,swvel_time,indices_interp,     &
-     &  imf_ncfile,saber_ncfile,tidi_ncfile,sech_nbyte, amie_ibkg, &
-     &  seeflux, amienh, amiesh 
-
-
-!-------------------------------------------------------------------------------
-
-if( .not. file_exist(file_name)) then
-   write(string1,*) trim(file_name),' not available.'
-   call error_handler(E_ERR,'read_TIEGCM_namelist',string1,source,revision,revdate)
-endif
-
-call error_handler(E_MSG,'read_TIEGCM_namelist:','reading namelist ['//trim(file_name)//']')
-
-! Read the namelist entry tgcm_input from tiegcm.nml
-! TJH It would be nice to read the namelist and skip all the ';' in column 1.
-! Basically, we are just getting the value of f10.7 and saving it.
-
-call find_namelist_in_file('tiegcm.nml', 'tgcm_input', iunit)
-read(iunit, nml = tgcm_input, iostat = io)
-call check_namelist_read(iunit, io, 'tgcm_input')
-
-if (step >= daysec) then
-    time_step_days    = int(step/daysec)
-    time_step_seconds = mod(step,daysec)
-else
-    time_step_days    = 0
-    time_step_seconds = step
-endif
-
-if (do_output() .and. (debug > 1)) then
-   write(string1,*) '..  tiegcm time_step_days    is ',time_step_days
-   write(string2,*) 'tiegcm time_step_seconds is ',time_step_seconds
-   call error_handler(E_MSG,'read_TIEGCM_namelist:',string1,text2=string2)
-endif
-
-f10_7 = f107  ! save this in module storage
-
-if (do_output() .and. (debug > 1)) then
-   write(string1,*) '..  f107 from tiegcm.nml     is ',f107
-   call error_handler(E_MSG,'read_TIEGCM_namelist:',string1)
-endif
-
-end subroutine read_TIEGCM_namelist
-
-
-!-------------------------------------------------------------------------------
-
-
 subroutine read_TIEGCM_definition(file_name)
```

Comment:
also eventually could be converted to using the dart netcdf util routines.

Reply:
ditto

Reply:
done 8445726fee511fa19ecd2630a559a2dc2bdbd691